packages/react-frontend/src/routes/Album.jsx

```diff
@@ -75,17 +108,37 @@ export default function Album() {
         );
       }
 
+      // Filter by average rating
+      if (selectedRating !== "") {
+        filteredData = filteredData.filter((album) => {
+          const albumRating = averageRatings[album._id] || 0;
+          return albumRating === parseFloat(selectedRating);
+        });
+      }
+
       setFilteredAlbums(filteredData);
       setLoading(false); // Finish loading
     };
     filterAlbums();
-  }, [selectedGenre, selectedYear, albums]);
+  }, [selectedGenre, selectedYear, selectedRating, albums]);
 
   return (
     <div>
       <Navbar withLogo={true} />
       <div className="content">
         <div className="filter-container">
+        <select
+            id="ratingFilter"
+            value={selectedRating}
+            onChange={(e) => setSelectedRating(e.target.value)}
+          >
+            <option value="">RATING</option>
```

Comment:
Why is the value just "". Will this eventually be something?

Reply:
The value is just "" because it indicates when a filter is not chosen, so it won't apply to the filtered albums. I tested it without the `value=""` parts and it doesn't work, when I apply all the filters, then unfilter, it doesn't show any albums afterwards.